### PR TITLE
NUI comma api intergration

### PR DESCRIPTION
**Pull request by ayman-saleh**
Prototyping nui comma api change

To run:
Generate `Makefile` with `qmake`,
Run `tools/lib/auth.py` to authentificate,
Run `make`
Run `./nui`

### CODE REVIEW

File: tools/nui/main.cpp [hunk 1]
```diff
@@ -55,6 +64,23 @@ Window::Window(QString route_, int seek) : route(route_) {
   connect(unlogger, SIGNAL (elapsed()), this, SLOT (update()));
   thread->start();
 
+  QString settings;
+  QFile file;
+  file.setFileName("/home/sal-smallboi-2/openpilot/tools/nui/routes.json");
```
Comment (geohot) [inline tools/nui/main.cpp]: well this won't work on my computer...

Comment (ayman-saleh) [reply tools/nui/main.cpp]: Changed it to just `routes.json`

**Comment by geohot** [comment]
Revert https://github.com/commaai/openpilot/commit/7f6c9ebaf79bed30c8f291b7499e8fe254050973#diff-af100c3f08db61c82c6952f00f099be4 and it works!

**Comment by ayman-saleh** [comment]
Added generalized changes

**Comment by geohot** [comment]
Why is routes.json committed?

File: tools/nui/main.cpp [hunk 2]
```diff
@@ -55,6 +64,23 @@ Window::Window(QString route_, int seek) : route(route_) {
...
+  file.setFileName("routes.json");
+  file.open(QIODevice::ReadOnly | QIODevice::Text);
+  settings = file.readAll();
+  file.close();
+    
+  QJsonDocument sd = QJsonDocument::fromJson(settings.toUtf8());
+  qWarning() << sd.isNull(); // <- print false :)
+  QJsonObject sett2 = sd.object();
+
+  QJsonArray camera_paths = sett2.value("camera").toArray();
```
Comment (geohot) [inline tools/nui/main.cpp]: Why create these locally?

File: tools/nui/FileReader.cpp [hunk 3]
```diff
@@ -10,7 +9,9 @@ FileReader::FileReader(const QString& file_) : file(file_) {
 void FileReader::process() {
   timer.start();
   // TODO: Support reading files from the API
```
Comment (geohot) [inline tools/nui/FileReader.cpp]: Remove TODO

Comment (ayman-saleh) [reply tools/nui/FileReader.cpp]: Right now, code only supports using public api, should I add a check to use either use old method or public api?

File: tools/nui/nui [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+if [ $# -gt 0 ]; then
+    if [ "$2" = use_api ]; then
+        python get_files_comma_api.py $1 && ./_nui use_api
+    else
```
Comment (geohot) [inline tools/nui/nui]: use an env var for internal, say INTERNAL=1

**Pull request merged**